pkg/precision/README.md

```diff
@@ -0,0 +1,34 @@
+## Problem
+
+- **Extensive heap usage**: the current implementation involves significant heap usage due to frequent allocations when operating on `decimal.Decimal` or `big.Int`. When these operations are chained, the inefficiency multiplies.
+- **Order book servicing**: representing the order book as a binary tree demands continuous balancing, adding computational overhead.
+- **Time complexity**: searching through the order book is log-linear $O(log(n))$ with respect to the size of the order book, making it less efficient as the order book grows.
+
+## Concept
+
+If the price of SHIBUSDT is *`0.00000792`*, a trade would mentally operate on it as just *`792`* and not *`zero, dot, 5 zeros, 792`*. The same idea can be used to streamline the representation of prices by using a fixed number of significant digits – so the price is broken down to 8 significant digits – *`0000 0792`*. 
```

Comment:
You are mixing the concept of significant digits and precision, did you mean a fixed precision?
Having the same precision between 2 numbers is mandatory to be able to compare them.
That's what a decimal library does for you, it stores the number as a []bytes with a precision number.